modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -3145,6 +3145,63 @@ void forceCloseCaches(ExchangeActions exchActions) {
         processCacheStopRequestOnExchangeDone(exchActions);
     }
 
+    /**
+     * Blocks cache gateways before rolling back transactions that were affected by Exchange,
+     * which in turn was caused by the destroyCache call for these caches.
+     * This should be done in order to prohibit enlisting new entries into optimistic transactions.
+     *
+     * @param exchActions Change requests.
+     */
+    private void blockCacheGatewaysOnExchangeDone(ExchangeActions exchActions) {
```

Comment:
This is old version of logic, also this method is not used. We should remove it.

Reply:
Shame on me! Will remove.